drivers/SmartThings/zigbee-window-treatment/src/aqara/curtain/init.lua

```diff
@@ -0,0 +1,234 @@
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local aqara_utils = require "aqara/aqara_utils"
+local zcl_commands = require "st.zigbee.zcl.global_commands"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+local AnalogOutput = clusters.AnalogOutput
+
+local deviceInitialization = capabilities["stse.deviceInitialization"]
+local deviceInitializationId = "stse.deviceInitialization"
+local setInitializedStateCommandName = "setInitializedState"
+
+local reverseCurtainDirectionPreferenceId = "stse.reverseCurtainDirection"
+local softTouchPreferenceId = "stse.softTouch"
+
+local INIT_STATE = "initState"
+local INIT_STATE_INIT = "init"
+local INIT_STATE_OPEN = "open"
+local INIT_STATE_CLOSE = "close"
+local INIT_STATE_REVERSE = "reverse"
+
+local PREF_INITIALIZE = "\x00\x01\x00\x00\x00\x00\x00"
+local PREF_SOFT_TOUCH_OFF = "\x00\x08\x00\x00\x00\x01\x00"
+local PREF_SOFT_TOUCH_ON = "\x00\x08\x00\x00\x00\x00\x00"
+
+local function window_shade_level_cmd(driver, device, command)
+  aqara_utils.shade_level_cmd(driver, device, command)
+end
+
+local function window_shade_open_cmd(driver, device, command)
+  aqara_utils.shade_open_cmd(driver, device, command)
+end
+
+local function window_shade_close_cmd(driver, device, command)
+  aqara_utils.shade_close_cmd(driver, device, command)
+end
+
+local function window_shade_pause_cmd(driver, device, command)
+  aqara_utils.shade_pause_cmd(driver, device, command)
+end
+
+local function write_initialize(device)
+  aqara_utils.write_pref_attribute(device, PREF_INITIALIZE)
+end
+
+local function setInitializationField(device, value)
+  device:set_field(INIT_STATE, value)
+end

```

Comment:
this is only used in the next function and then you go back to calling `device:set_field`

I recommend using `device:set_field` everywhere

Reply:
I'll edit and push.